ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt

```diff
@@ -0,0 +1,152 @@
+    fun addCommandAndSignatories(command: Command, vararg signatories: PublicKey): UtxoTransactionBuilder {
+        return addCommandAndSignatories(command, signatories.toSet())
+    }
+
+    /**
+     * Adds an input state to the current [UtxoTransactionBuilder].
+     *
+     * @param stateAndRef The [StateAndRef] of the input state to add to the current [UtxoTransactionBuilder].
+     * @return Returns a [UtxoTransactionBuilder] including the additional input state.
+     */
+    fun addInputState(stateAndRef: StateAndRef<*>): UtxoTransactionBuilder
+
+    /**
+     * Adds a reference input state to the current [UtxoTransactionBuilder].
+     *
+     * @param stateAndRef The [StateAndRef] of the reference input state to add to the current [UtxoTransactionBuilder].
+     * @return Returns a [UtxoTransactionBuilder] including the additional reference input state.
+     */
+    fun addReferenceInputState(stateAndRef: StateAndRef<*>): UtxoTransactionBuilder
+
+    /**
+     * Adds an output state to the current [UtxoTransactionBuilder].
+     *
+     * @param contractState The [ContractState] to add to the current [UtxoTransactionBuilder].
+     * @return Returns a [UtxoTransactionBuilder] including the additional output state.
+     */
+    fun addOutputState(contractState: ContractState): UtxoTransactionBuilder
+
+    /**
+     * Adds an output state to the current [UtxoTransactionBuilder].
+     *
+     * @param contractState The [ContractState] to add to the current [UtxoTransactionBuilder].
+     * @param encumbrance The index of an associated, encumbered state, or null if no encumbrance applies to the associated transaction state.
+     * @return Returns a [UtxoTransactionBuilder] including the additional output state.
+     */
+    fun addOutputState(contractState: ContractState, encumbrance: Int?): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid until the specified [Instant], tending towards negative infinity.
+     *
+     * @param until The [Instant] until which the transaction time window is valid.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowUntil(until: Instant): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid between the specified [Instant] values.
+     *
+     * @param from The [Instant] from which the transaction time window is valid.
+     * @param until The [Instant] until which the transaction time window is valid.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowBetween(from: Instant, until: Instant): UtxoTransactionBuilder
+
+    /**
+     * Signs the transaction with any required signatories that belong to the current node.
+     *
+     * @return Returns a [UtxoSignedTransaction] with signatures for any required signatories that belong to the current node.
+     */
+    fun sign(): UtxoSignedTransaction
```

Comment:
It's called `signInitial` on `ConsensualTransactionBuilder`, we should plan on aligning them.

Reply:
Ye I think `sign` itself sounds right.

Reply:
So long as people understand that this is a partially signed transaction, not a fully complete one